ncm-freeipa/src/main/pan/components/freeipa/schema.pan

```diff
@@ -0,0 +1,167 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+declaration template components/${project.artifactId}/schema;
+
+include 'pan/types';
+include 'quattor/types/component';
+
+@{ group members configuration }
+type component_${project.artifactId}_member = {
+    @{(minimal) user group members}
+    'user' ? string[]
+};
+
+@{ group configuration }
+type component_${project.artifactId}_group = {
+    @{group ID number}
+    'gidnumber' : long(0..)
+    @{group members}
+    'members' ? component_${project.artifactId}_member
+};
+
+@{ service configuration }
+type component_${project.artifactId}_user = {
+    @{user ID number}
+    'uidnumber' : long(0..)
+    @{last name}
+    'sn' : string
+    @{first name}
+    'givenname' : string
+    @{group name (must be a configured group to retrieve the gid)}
+    'group' ? string with exists('/software/components/${project.artifactId}/server/groups/'+SELF)
+    @{homedirectory}
+    'homedirectory' ? string
+    @{gecos}
+    'gecos' ? string
+    @{loginshell}
+    'loginshell' ? string with match(SELF, '^/bin/(ba|c)sh$')
```

Comment:
The regex seems a bit restrictive. /etc/shells on my box is:
```
/bin/bash
/bin/csh
/bin/ksh
/bin/sh
/bin/tcsh
/bin/zsh
```
which makes for a pretty horrible regex!